mllib/src/main/scala/org/apache/spark/mllib/linalg/BLAS.scala

```diff
@@ -266,8 +264,6 @@ private[spark] object BLAS extends Serializable with Logging {
    * @param C the resulting matrix C. Size of m x n.
```

Comment:
What will happen if `C` is transposed? This needs some documentation. I think we can throw an exception if `C` is transposed in this PR.